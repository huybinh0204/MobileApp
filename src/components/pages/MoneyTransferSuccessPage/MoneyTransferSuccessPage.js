import { useNavigation, useRoute } from '@react-navigation/native';
import React from 'react';
import { useDispatch } from 'react-redux';
import { MainButton } from '_components/atoms';
import { TransferDetailsItem } from '_components/molecules';
import { SecondaryScreenLayout } from '_components/organisms';
import { ICONS, TRANSFER_TYPES } from '_constants';
import strings from '_localization';
import { AuthenticationActions } from '_store/authentication';
import { TransferActions } from '_store/transfer';
import { normalize } from '_utilities/screen';
import {
  ButtonContainer,
  Icon,
  InstantTransferDisclaimer,
  MoneyTransferSuccessContainer,
  Title,
} from './MoneyTransferSuccessPage.styles';

const MoneyTransferSuccessPage = () => {
  const dispatch = useDispatch();
  const { params } = useRoute();
  const { popToTop } = useNavigation();

  const { transferAmount, transferFee, transferFrom, transferTo, transferMethod } = params;
  const isInstantTransfer = transferMethod === TRANSFER_TYPES.INSTANT_OUT;

  const handlePress = () => {
    dispatch(TransferActions.resetTransfer());
    dispatch(AuthenticationActions.setIsSignIn(true));
    popToTop();
  };

  return (
    <SecondaryScreenLayout showBackButton={false} testID="MoneyTransferSuccessPage">
      <MoneyTransferSuccessContainer>
        <Icon icon={ICONS.checkmarkLines} width={normalize(80)} height={normalize(80)} />
        <Title>{strings.fund.success.title}</Title>
        <TransferDetailsItem
          label={strings.moneyMovement.transferDetails.transferAmount}
          value={`${strings.currencySymbol}${transferAmount.toFixed(2)}`}
        />
        {isInstantTransfer && (
          <TransferDetailsItem
            label={strings.moneyMovement.transferDetails.fee}
            value={`${strings.currencySymbol}${transferFee}`}
          />
        )}
        <TransferDetailsItem
          label={strings.moneyMovement.transferDetails.from}
          value={transferFrom}
        />
        <TransferDetailsItem label={strings.moneyMovement.transferDetails.to} value={transferTo} />
        <TransferDetailsItem
          isLastItem
          label={strings.moneyMovement.transferDetails.method}
          value={
            isInstantTransfer
              ? strings.moneyMovement.transferDetails.instantTransfer
              : strings.moneyMovement.transferDetails.standardTransfer
          }
        />
        <ButtonContainer>
          {isInstantTransfer && (
            <InstantTransferDisclaimer>
              {strings.moneyMovement.transferDetails.instantTransferDisclaimer}
            </InstantTransferDisclaimer>
          )}
          <MainButton onPress={handlePress} testID="moneyTransferSuccessButton">
            {strings.fund.success.gotIt}
          </MainButton>
        </ButtonContainer>
      </MoneyTransferSuccessContainer>
    </SecondaryScreenLayout>
  );
};

export default MoneyTransferSuccessPage;
