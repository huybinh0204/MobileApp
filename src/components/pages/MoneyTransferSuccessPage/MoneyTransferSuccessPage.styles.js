import styled from '@emotion/native';
import { BaseText, IconSvg } from '_components/atoms';

export const MoneyTransferSuccessContainer = styled.View(({ theme }) => ({
  flex: 1,
  alignItems: 'center',
  padding: theme.spacing.m,
}));

export const Icon = styled(IconSvg)(({ theme }) => ({
  marginBottom: theme.spacing.m,
}));

export const Title = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.m,
  textAlign: 'center',
}));

export const ButtonContainer = styled.View({
  alignSelf: 'stretch',
  marginTop: 'auto',
});

export const InstantTransferDisclaimer = styled(BaseText)(({ theme }) => ({
  ...theme.typography.legal,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));
