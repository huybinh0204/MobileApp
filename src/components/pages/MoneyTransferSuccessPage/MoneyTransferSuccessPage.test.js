import { useNavigation } from '@react-navigation/native';
import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { useDispatch } from 'react-redux';
import { TRANSFER_TYPES } from '_constants';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import MoneyTransferSuccessPage from './MoneyTransferSuccessPage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('MoneyTransferSuccessPage', () => {
  let params;
  const mockedDispatch = jest.fn();
  const mockedNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useDispatch.mockReturnValue(mockedDispatch);
    useNavigation.mockReturnValue(mockedNavigation);

    params = {
      transferAmount: 9.75,
      transferFee: 0.25,
      transferFrom: 'Kinly Account',
      transferTo: 'American Express',
      transferMethod: TRANSFER_TYPES.INSTANT_OUT,
    };
  });

  it('should match snapshot', () => {
    const { toJSON } = render(
      <ComponentWithProviders component={MoneyTransferSuccessPage} params={params} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate back to snapshot', () => {
    const { getByTestId } = render(
      <ComponentWithProviders component={MoneyTransferSuccessPage} params={params} />
    );

    const continueButton = getByTestId('moneyTransferSuccessButton');

    expect(continueButton).toBeEnabled();

    fireEvent.press(continueButton);

    expect(mockedNavigation.popToTop).toHaveBeenCalled();
  });
});
