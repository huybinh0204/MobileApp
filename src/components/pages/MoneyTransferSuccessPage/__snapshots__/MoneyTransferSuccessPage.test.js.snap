// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyTransferSuccessPage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="MoneyTransferSuccessPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        />
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <SvgMock
                              height={145}
                              style={
                                Array [
                                  Object {
                                    "marginBottom": 58,
                                  },
                                  undefined,
                                ]
                              }
                              width={145}
                            />
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 36,
                                    "lineHeight": 50.5,
                                    "marginBottom": 58,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Your money is on its way!
                            </Text>
                            <View
                              isLastItem={false}
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "borderBottomColor": "#F2F3F4",
                                    "borderBottomWidth": 1,
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "marginBottom": 0,
                                    "paddingVertical": 29,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Youâ€™ll transfer:
                              </Text>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                $9.75
                              </Text>
                            </View>
                            <View
                              isLastItem={false}
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "borderBottomColor": "#F2F3F4",
                                    "borderBottomWidth": 1,
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "marginBottom": 0,
                                    "paddingVertical": 29,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Instant Transfer Fee:
                              </Text>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                $0.25
                              </Text>
                            </View>
                            <View
                              isLastItem={false}
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "borderBottomColor": "#F2F3F4",
                                    "borderBottomWidth": 1,
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "marginBottom": 0,
                                    "paddingVertical": 29,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                From:
                              </Text>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Kinly Account
                              </Text>
                            </View>
                            <View
                              isLastItem={false}
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "borderBottomColor": "#F2F3F4",
                                    "borderBottomWidth": 1,
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "marginBottom": 0,
                                    "paddingVertical": 29,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                To:
                              </Text>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                American Express
                              </Text>
                            </View>
                            <View
                              isLastItem={true}
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "stretch",
                                    "borderBottomColor": "transparent",
                                    "borderBottomWidth": 1,
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "marginBottom": 29,
                                    "paddingVertical": 29,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Arrival:
                              </Text>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Instant
                              </Text>
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "marginTop": "auto",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#5F6A77",
                                      "fontFamily": "Poppins",
                                      "fontSize": 20,
                                      "lineHeight": 28,
                                      "marginBottom": 29,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Most Instant Transfers are instant but can take up to 30 minutes depending on your bank.
                              </Text>
                              <View
                                accessibilityState={
                                  Object {
                                    "disabled": false,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "#3D3397",
                                    "borderRadius": 10,
                                    "opacity": 1,
                                    "paddingHorizontal": 45.5,
                                    "paddingVertical": 31,
                                  }
                                }
                                testID="moneyTransferSuccessButton"
                              >
                                <Text
                                  allowFontScaling={false}
                                  disabled={false}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "textAlign": "center",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  Got it
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
